File: portfolio/src/services/view_count.py
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()
VIEW_COUNT_FILE = 'view_counts.txt'

class PageView(BaseModel):
    page: str
    count: int

def update_view_count(page):
    view_counts = read_view_counts()
    view_counts[page] = view_counts.get(page, 0) + 1
    write_view_counts(view_counts)

def get_view_count(page):
    view_counts = read_view_counts()
    return view_counts.get(page, 0)

def read_view_counts():
    try:
        with open(VIEW_COUNT_FILE, 'r') as file:
            view_counts = {}
            for line in file:
                page, count = line.strip().split(':')
                view_counts[page] = int(count)
            return view_counts
    except FileNotFoundError:
        return {}

def write_view_counts(view_counts):
    with open(VIEW_COUNT_FILE, 'w') as file:
        for page, count in view_counts.items():
            file.write(f"{page}:{count}\n")

File: astro-site/src/utils/viewCount.ts
import fs from 'fs';
import path from 'path';

const VIEW_COUNT_FILE = path.join(process.cwd(), '..', 'view_counts.txt');

/**
 * Read view counts from file
 * Returns an object with page paths as keys and counts as values
 */
export function readViewCounts(): Record<string, number> {
  try {
    const data = fs.readFileSync(VIEW_COUNT_FILE, 'utf-8');
    const viewCounts: Record<string, number> = {};

    const lines = data.trim().split('\n');
    for (const line of lines) {
      if (line.trim()) {
        const [page, count] = line.split(':');
        viewCounts[page] = parseInt(count, 10);
      }
    }

    return viewCounts;
  } catch (error) {
    // File doesn't exist or can't be read, return empty object
    return {};
  }
}

/**
 * Write view counts to file
 */
export function writeViewCounts(viewCounts: Record<string, number>): void {
  const lines = Object.entries(viewCounts)
    .map(([page, count]) => `${page}:${count}`)
    .join('\n');

  fs.writeFileSync(VIEW_COUNT_FILE, lines + '\n', 'utf-8');
}

/**
 * Update (increment) view count for a page
 */
export function updateViewCount(page: string): void {
  const viewCounts = readViewCounts();
  viewCounts[page] = (viewCounts[page] || 0) + 1;
  writeViewCounts(viewCounts);
}

/**
 * Get view count for a specific page
 */
export function getViewCount(page: string): number {
  const viewCounts = readViewCounts();
  return viewCounts[page] || 0;
}
